builder/context.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package builder
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/Azure/acr-builder/util"
+
+	"github.com/Azure/acr-builder/baseimages/scanner/models"
+	"github.com/google/uuid"
+)
+
+const (
+	defaultScannerImage = "scanner"
+)
+
+var (
+	dependenciesRE = regexp.MustCompile(`^(\[{"image.*?\])$`)
+)
+
+// getDockerRunArgs populates the args for running a Docker container.
+func (b *Builder) getDockerRunArgs(volName string, stepID string, stepWorkDir string) []string {
+	args := []string{"docker", "run"}
+
+	if rmContainer {
+		args = append(args, "--rm")
+	}
+
+	args = append(args,
+		"--name", fmt.Sprintf("acb_step_%s", stepID),
+		"--volume", volName+":"+containerWorkspaceDir,
+
+		// Mount home
+		"--volume", util.GetDockerSock(),
+		"--volume", homeVol+":"+homeWorkDir,
+		"--env", "HOME="+homeWorkDir,
```

Comment:
Is HOME the reserved environment parameter for the pipeline?

Reply:
`HOME=/blah/blah` sets `$HOME=/blah/blah` so when docker CLI or any other program says something like `~/.docker/config.json` it will resolve to `/blah/blah/.docker/config.json`

Reply:
windows support $HOME?